src/cpp/rtps/security/SecurityManager.cpp

```diff
@@ -813,11 +802,7 @@ bool SecurityManager::on_process_handshake(
     assert(remote_participant_info->handshake_handle_ != nullptr);
 
     // Remove previous change
-    if (remote_participant_info->event_ != nullptr)
-    {
-        delete remote_participant_info->event_;
-        remote_participant_info->event_ = nullptr;
-    }
+    remote_participant_info->event_.reset();
```

Comment:
Do we have to be constantly destroying and creating the timed event?
Could we always create it on the constructor of `AuthenticationInfo`, and just cancel it here, and restart it below?